clientconn.go

```diff
@@ -1824,7 +1824,7 @@ func (cc *ClientConn) initAuthority() error {
 	} else if auth, ok := cc.resolverBuilder.(resolver.AuthorityOverrider); ok {
 		cc.authority = auth.OverrideAuthority(cc.parsedTarget)
 	} else if strings.HasPrefix(endpoint, ":") {
-		cc.authority = "localhost" + endpoint
+		cc.authority = "localhost" + encodeAuthority(endpoint)
```

Comment:
Is this a breaking change? Should this be release noted?

Reply:
It's a behavior change.  Unclear if it's breaking; I would call it a bug fix.  I'll mention it in the notes.

Reply:
Added:

Also, properly percent-encode the port (if needed, which is very unlikely) when the target string omits the hostname and only specifies a port (`grpc.NewClient(":<port-number-or-name>")`).